app/routes/upload.step2.tsx

```diff
@@ -1,15 +1,52 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { redirect, useNavigation } from '@remix-run/react'
 import { useState } from 'react'
 import { Button } from '~/components/ui/button'
 import { Input } from '~/components/ui/input'
 import { Textarea } from '~/components/ui/textarea'
+import { formCookie } from '~/lib/cookies'
+
+export const action = async ({ request }: ActionFunctionArgs) => {
+  // Step 1: Get form data
+  const formData = await request.formData()
+  const jobTitle = formData.get('jobTitle') as string
+  const companyName = formData.get('companyName') as string
+  const jobDescription = formData.get('jobDescription') as string
+
+  // Step 2: Retrieve the current cookie value
+  const cookieHeader = request.headers.get('Cookie')
+  const existingData = (await formCookie.parse(cookieHeader)) || {}
```

Comment:
**issue**: Consider removing console.log statements in production code to avoid unnecessary logging.